osu.Game.Tests/Chat/MessageFormatterTests.cs

```diff
@@ -428,23 +428,23 @@ public void TestLinkComplex()
             Assert.AreEqual(5, result.Links.Count);
 
             Link f = result.Links.Find(l => l.Url == "https://osu.ppy.sh/wiki/wiki links");
-            Assert.AreEqual(44, f.Index);
+            Assert.AreEqual(44, f!.Index);
```

Comment:
We haven't enable NRT, so the `!` means nothing for the compiler. Asserting f isn't null should be better approach.

Reply:
Resharper analysis != NRT, so it means something here. It means resharper won't complain and will let an exception occur if it's null. This won't ever be the case but since it's a test it's effectively the same as an assertion imo.

Reply:
What does this mean? Are there any alternatives to `!`? Feels quite foreign.